### PR TITLE
Fix a bug with blockview

### LINKED ISSUE #1247: block_view2 with end_view=None doesn't work as intended
The intention behind `end_view=None` seems to be to read until the end of file, however, it ends up not updating the file_kw in the loop so that the first kw is pushed onto the list every time. Can be reproduced with

```python
with openResdataFile("test-data/local/ECLIPSE/simple/SIMPLE.EGRID") as file:
    view = file.block_view2(None, None, 0)
    print(list(view)) # Should include all keywords, but just contains "FILEHEAD" seven times.
```

### PR DESCRIPTION
**Pull request by eivindjahren**
**Issue**
Resolves #1247

### CODE REVIEW

File: python/resdata/resfile/rd_file.py [hunk 1]
```diff
@@ -222,7 +244,44 @@ def block_view(self, kw, kw_index):
             "Index out of range, must be in [0, %d), was %d." % (ls, kw_index)
         )
 
-    def block_view2(self, start_kw, stop_kw, start_index):
+    def block_view2(
```
Comment (MagnusSletten) [inline python/resdata/resfile/rd_file.py:247]: block_view1 and block_view2 is maybe not the clearest naming scheme?

**Review by MagnusSletten — APPROVED** [review]
Looks good! The bug is clearly addressed and the accompanying tests cover the edge cases well. The extra documentation is also nice. 

One small optional point about the python naming scheme. It's not blocking.

File: python/resdata/resfile/rd_file.py (hunk 1 above)
Comment (eivindjahren) [reply python/resdata/resfile/rd_file.py:247]: 100%, however, for now we should avoid making too large changes to the interface.
